ui/Surface.js

```diff
@@ -191,6 +181,15 @@ Surface.Prototype = function() {
         this.clipboard.attach(el);
       }
     }
+    
+    if (this.props.overlay) {
+      el.append(
```

Comment:
Is the overlay thing already part of this work actually?
I would say you remove that here and get this PR merged earlier.